packages/atlas/src/providers/assets/assets.hooks.ts

```diff
@@ -39,3 +40,74 @@ export const useMemberAvatar = (member?: BasicMembershipFieldsFragment | null):
 
   return { url: null, isLoadingAsset: true }
 }
+
+export const useSubtitlesAssets = (subtitles?: SubtitlesFieldsFragment[] | null) => {
```

Comment:
I think this hooks is overly complicated, it's very hard for me to understand what is happening.

There are also some issues:
1. You use full `assets` and `pendingAssets` state. This hook will force a re-render whenever something in those objects changes which may be quite often. We need to be careful to not reduce performance of `VideoView`
2. This is placed inside the assets provider but also contains some logic about display of subtitles (CC, language, etc). We should separate concerns of resolving assets and creating lists of subtitles.

Take a look at my PR to your branch: https://github.com/rafalpawlow/atlas/pull/9 I simplified this hook with couple of scoped store accessors and single useEffect. I managed to reduce the number of VideoView renders by around 10 (it's still stupidly high, but that's because of other unoptimized code). Please review, I think it works properly.